### DISCUSSION TITLE
Download and Queue Management for Models

### DISCUSSION THREAD

**Discussion by wyxos** [BlafKing/sd-civitai-browser-plus#244] [Ideas]
**Download and Queue Management for Models**
- **Issue**: When multiple models are being downloaded, the interface lacks a "Download All Selected" or "Add All Selected to Queue" option. This limitation requires users to add models to the download queue individually, which can be time-consuming and cumbersome.
- **Expected Behavior**: Introduce a "Download All Selected" or "Add All Selected to Queue" button that allows users to efficiently manage multiple models for download or queuing.

**Selective Addition to Queue**
- **Issue**: If multiple models are selected and one of them is clicked on, the "Add to Queue" option only adds the model that was clicked, ignoring the others that were also selected.
- **Expected Behavior**: The "Add to Queue" function should recognize all selected models, adding them to the queue simultaneously rather than just the one that was last clicked.

**Search Functionality and Relevance Filtering**
- **Issue**: The search feature appears to be using an outdated version of the search engine. It lacks the ability to filter results by "relevance," and certain models that are available when searching directly on the site do not appear when browsing through the extension, such as models containing reference to "anime style."
- **Expected Behavior**: The search functionality should incorporate the latest search engine advancements, including the ability to filter results by relevance and ensure that all models available on the site are also accessible through the extension.

**Comment by BlafKing** [comment]
Thanks for writing in!

The first two points you wrote can definitely be fixed, I'll add those to the to-do list!

About your last point, this is unfortunately not possible due to this extension having to use the public REST API from CivitAI.
It's not an outdated version of the search engine, but the one that's open to the public, the CivitAI site uses an internal API which is only accessible through their website which is where the differing features come from, so unfortunately the functionality you requested can only be added by CivitAI themselves.

**Comment by wyxos** [comment]
Thanks for explaining that – I had a feeling it might be something to do with CivitAI's API limitations. Good to know you'll tackle the first two points. I'll see if CivitAI might consider making their internal search features more accessible. Looking forward to the updates!
